api/envoy/extensions/filters/http/dynamic_forward_proxy/v3/dynamic_forward_proxy.proto

```diff
@@ -27,6 +27,12 @@ message FilterConfig {
   // <envoy_v3_api_field_extensions.clusters.dynamic_forward_proxy.v3.ClusterConfig.dns_cache_config>`.
   common.dynamic_forward_proxy.v3.DnsCacheConfig dns_cache_config = 1
       [(validate.rules).message = {required: true}];
+
+  // When this flag is set, the filter will add the resolved upstream address in the filter
+  // state. The state should be saved with key
+  // `envoy.rbac.matcher.upstream_address_set` (See
```

Comment:
Shephards get final call but I can imagine this key being useful for other filters, so I wonder if it should have naming based on the DFP (originator) rather than RBAC (current consumer)

Reply:
Yes, agree, although this points to a confusion in how filter state is supposed to be used; is it addressed by sender or receiver? It coudl be both, but then there is potential for confusion on ownership of the rest of the namespace. Maybe we need something like `envoy.<filter name>.<source|sink>.<attribute>`, but I think it's overkill.

The expedient thing might be to say that things that are augmenting the set of attributes that are consumed by filters using attributes, e.g. in ACL, in Wasm, have their own special namespace to add to, `envoy.request.upstream_address` in this case.

Open to other suggestions in the bike shed.

Reply:
>  is it addressed by sender or receiver?

I think both cases exist in the codebase. For example, there's one that tcp_proxy will read to override which cluster to route to; any network filter can set that (`Envoy::TcpProxy::PerConnectionCluster::key()`, which is `"envoy.tcp_proxy.cluster"`.

In this case, the DFP is broadcasting this data, and many things could in theory consume it.

So I don't think always naming it in one way or the other will work well. But maybe there are just these two cases (specific consumer, vs broadcaster), and we can name based on which of those cases it is?

Reply:
@htuch @ggreenway I have changed the key name to `envoy.request.upstream_address_set` . IMHO the key should reflect the use case/purpose here which is `rbac` filter.